File: Java/src/odev/TekTaktik.java
/**
*
* @author Sude Çakmak / sude.cakmak1@ogr.sakarya.edu.tr
* @since 30.05.2023
* <p>
* Taktik için kullanılan sınıflardan biri (1-1000 arası rastgele değer döndürür)
* </p>
*/
package odev;

import java.util.Random;

public class TekTaktik extends Taktik {   //Taktik abstract sınıfından kalıtım alıyor
	public TekTaktik(int miktar) {
		super(miktar);   //abstract sınıfın yapıcısını çağırıyor
	}
	
	@Override  //abstract sınıftaki metot override ediliyor
	public int Savas() {    // 1-1000 arasında değer döndüren metot
		Random r = new Random();   //dönen değere rastgelelik katması için
		int rastgele = this.baslangic_populasyon*7;     //bu metoda özgü
		int rand = r.nextInt(2000);
		rastgele += rand;
		rastgele = (rastgele % 1000) ;   // sayının 1000'e modu alınarak 0-1000 arası değer gelmesi sağlanıyor
		if(1 < rastgele && rastgele < 1000) return rastgele;
		else return 7;  //0 ya da 1 gelme ihtimaline karşılık (1-1000 arasında sayı istendiği için 1 ve 1000 dahil değil)
	}
}

File: C/src/TekTaktik.c
/** 
* @file ./src/TekTaktik.cpp
* @description TekTaktik yapısının kaynak dosyası
* @course 1. öğretim B grubu
* @assignment 2. Ödev
* @date 18.05.2023
* @author Sude Çakmak sude.cakmak1@ogr.sakarya.edu.tr
*/

#include "TekTaktik.h"

TekTaktik TekTakOlustur(int miktar){  //kurucu fonksiyon
	TekTaktik this;
	this = (TekTaktik)malloc(sizeof(struct TEKTAKTIK));
	this->super = new_Taktik(miktar);  //kalıtım alınan yapı
	this->super->savas = &savas;
	this->super->kalitimyoket = &TekTakYoket;
	this->yoket = &TekTakYoket;
	return this;
}
int savas(const Taktik this, int aktif_populasyon){  // 1 ile 1000 arasında rastgele değer
	int deger = rand() + (this->baslangic_populasyon+aktif_populasyon)*3;
	deger = abs(deger);
	deger = deger % 1000;
	if(deger <= 1 ) return 7;  //0 ya da 1 gelirse diye
	else return deger;

}
void TekTakYoket(const TekTaktik this){  //yıkıcı fonksiyon
	if(this == NULL) return;
	this->super->yoket(this->super);
	free(this);
}